Windows: Import error after pip install

## Bug Report

-- edit: removed the checkboxes so that GitHub is no longer confused about open tasks --

-  Incorrect functionality

Describe the bug here. If it is a build issue, please note whether the issue was encountered via `cmake`, `pip`, or `tox`.

## To Reproduce

1. Operating System: Windows 10
2. Python 3.7.9, CMake installed
3. Example snippet that demonstrates the issue:
```python
ImportError: DLL load failed: The specified module could not be found.
```
4. OpenTimelineIO 0.13.0

## Expected Behavior

The import statement to work correctly.

## Logs
```python
Traceback (most recent call last):
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "c:\Users\User\.vscode\extensions\ms-python.python-2020.9.114305\pythonFiles\lib\python\debugpy\__main__.py", line 45, in <module>
    cli.main()
  File "c:\Users\User\.vscode\extensions\ms-python.python-2020.9.114305\pythonFiles\lib\python\debugpy/..\debugpy\server\cli.py", line 430, in main
    run()
  File "c:\Users\User\.vscode\extensions\ms-python.python-2020.9.114305\pythonFiles\lib\python\debugpy/..\debugpy\server\cli.py", line 267, in run_file
    runpy.run_path(options.target, run_name=compat.force_str("__main__"))
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "E:\Logiciels\Stax\Dev\stax\test_otio.py", line 4, in <module>
    import opentimelineio as otio
  File "e:\Logiciels\Stax\Dev\stax\.venv\lib\site-packages\opentimelineio\__init__.py", line 35, in <module>
    from . import (
  File "e:\Logiciels\Stax\Dev\stax\.venv\lib\site-packages\opentimelineio\opentime.py", line 1, in <module>
    from . _opentime import ( # noqa
ImportError: DLL load failed: The specified module could not be found.
```

More info: I see in your log that pybind11 is failing to find Python...

```
    -- Found PythonInterp: E:/Logiciels/Stax/Dev/blender-2.92.0-c13754e6475e-windows64/2.92/python/bin/python.exe (found version "3.7.7")
    CMake Error at src/deps/pybind11/tools/FindPythonLibsNew.cmake:158 (message):
      Python libraries not found
    Call Stack (most recent call first):
      src/deps/pybind11/tools/pybind11Tools.cmake:16 (find_package)
      src/deps/pybind11/CMakeLists.txt:33 (include)
```

Daily report:
I can compare the successful (windows python3.7.9) and the failing one (blender python 3.7.7).

The weels building `bdist_wheel` fails in both cases and falls back on the `install`.

It seems to be an issue in the `src\deps\pybind11\tools\FindPythonLibsNew.cmake`, the `PYTHON_LIBRARY` var is always deduced from the python's executable and infers filepaths which cannot be overriden only on win: https://github.com/pybind/pybind11/blob/8de7772cc72daca8e947b79b83fea46214931604/tools/FindPythonLibsNew.cmake#L184

These directories don't exist under Blenders python:
```
python/
|
|--bin/
  |--python.exe 
  |--python37.dll
|--DLLs/...
|--lib/...
```

Setting by hand directly in the code `PYTHON_LIBRARY` to the `lib/` dir works and the libs are correctly found, but leads to a new error:
```
CMake Error in src/py-opentimelineio/opentime-bindings/CMakeLists.txt:
      Imported target "pybind11::module" includes non-existent path

        "E:/Logiciels/Stax/Dev/blender-2.92.0-windows64/2.92/python/include"

      in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

      * The path was deleted, renamed, or moved to another location.

      * An install or uninstall procedure did not complete successfully.

      * The installation package was faulty and references files it does not
      provide.
```
Which looks like a similar issue, where the `include` dir is supposed to be there by default but doesn't exist on blender's python.

Trying to disable the `PYTHON_INCLUDE_DIR` (set empty) leads to these issues (sharing in case it pops up something to your mind):
```
WARNING: Target "_opentime" requests linking to directory "E:/Logiciels/Stax/Dev/blender-2.92.0-windows64/2.92/python/lib".  Targets may link only to libraries.  CMake is dropping the item.
...
C:\Users\Felix\AppData\Local\Temp\pip-req-build-_i66ief9\src\deps\pybind11\include\pybind11\detail/common.h(124,10): fatal error C1083: Cannot open include file: 'Python.h': No such file or directory
```

Hope the information is useful, will continue my cluedo tomorrow.

--- a/setup.py
+++ b/setup.py
@@ -97,8 +97,8 @@ def cmake_generate():
         '-DOTIO_PYTHON_INSTALL:BOOL=ON',
         '-DOTIO_CXX_INSTALL:BOOL=ON',
         '-DCMAKE_BUILD_TYPE=' + ('Debug' if _ctx.debug else 'Release'),
-        '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs:  https://pybind11.readthedocs.io/en/latest/compiling.html#findpython-mode
-    ]
+        '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs
+    ]  # https://pybind11.readthedocs.io/en/latest/compiling.html#findpython-mode
 
     python_inst_dir = get_python_lib()
 

--- a/setup.py
+++ b/setup.py
@@ -96,7 +96,8 @@ def cmake_generate():
         '-DPYTHON_EXECUTABLE=' + sys.executable,
         '-DOTIO_PYTHON_INSTALL:BOOL=ON',
         '-DOTIO_CXX_INSTALL:BOOL=ON',
-        '-DCMAKE_BUILD_TYPE=' + ('Debug' if _ctx.debug else 'Release')
+        '-DCMAKE_BUILD_TYPE=' + ('Debug' if _ctx.debug else 'Release'),
+        '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs
     ]
 
     python_inst_dir = get_python_lib()

--- a/setup.py
+++ b/setup.py
@@ -97,7 +97,7 @@ def cmake_generate():
         '-DOTIO_PYTHON_INSTALL:BOOL=ON',
         '-DOTIO_CXX_INSTALL:BOOL=ON',
         '-DCMAKE_BUILD_TYPE=' + ('Debug' if _ctx.debug else 'Release'),
-        '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs
+        '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs:  https://pybind11.readthedocs.io/en/latest/compiling.html#findpython-mode 
     ]
 
     python_inst_dir = get_python_lib()

--- a/setup.py
+++ b/setup.py
@@ -97,7 +97,7 @@ def cmake_generate():
         '-DOTIO_PYTHON_INSTALL:BOOL=ON',
         '-DOTIO_CXX_INSTALL:BOOL=ON',
         '-DCMAKE_BUILD_TYPE=' + ('Debug' if _ctx.debug else 'Release'),
-        '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs:  https://pybind11.readthedocs.io/en/latest/compiling.html#findpython-mode 
+        '-DPYBIND11_FINDPYTHON=ON',  # Smart tool to find python's libs:  https://pybind11.readthedocs.io/en/latest/compiling.html#findpython-mode
     ]
 
     python_inst_dir = get_python_lib()

